modules/file-system/parallelstore/templates/mount-daos.sh.tftpl

```diff
@@ -20,59 +20,55 @@ OS_VERSION=$(awk -F '=' '/VERSION_ID/ {print $2}' /etc/os-release | sed -e 's/"/
 OS_VERSION_MAJOR=$(awk -F '=' '/VERSION_ID/ {print $2}' /etc/os-release | sed -e 's/"//g' -e 's/\..*$//')
 
 if ! {
-	{ [[ "${OS_ID}" = "rocky" ]] || [[ "${OS_ID}" = "rhel" ]]; } && { [[ "${OS_VERSION_MAJOR}" = "8" ]] || [[ "${OS_VERSION_MAJOR}" = "9" ]]; } ||
-		{ [[ "${OS_ID}" = "ubuntu" ]] && [[ "${OS_VERSION}" = "22.04" ]]; } ||
-		{ [[ "${OS_ID}" = "debian" ]] && [[ "${OS_VERSION_MAJOR}" = "12" ]]; }
+	{ [[ "$${OS_ID}" = "rocky" ]] || [[ "$${OS_ID}" = "rhel" ]]; } && { [[ "$${OS_VERSION_MAJOR}" = "8" ]] || [[ "$${OS_VERSION_MAJOR}" = "9" ]]; } ||
+		{ [[ "$${OS_ID}" = "ubuntu" ]] && [[ "$${OS_VERSION}" = "22.04" ]]; } ||
+		{ [[ "$${OS_ID}" = "debian" ]] && [[ "$${OS_VERSION_MAJOR}" = "12" ]]; }
 }; then
-	echo "Unsupported operating system ${OS_ID} ${OS_VERSION}. This script only supports Rocky Linux 8, Redhat 8, Redhat 9, Ubuntu 22.04, and Debian 12."
+	echo "Unsupported operating system $${OS_ID} $${OS_VERSION}. This script only supports Rocky Linux 8, Redhat 8, Redhat 9, Ubuntu 22.04, and Debian 12."
 	exit 1
 
 fi
 
-# Parse local_mount, mount_options from argument.
-# Format mount-options string to be compatible to dfuse mount command.
-# e.g. "disable-wb-cache,eq-count=8" --> --disable-wb-cache --eq-count=8.
-for arg in "$@"; do
-	if [[ $arg == --access_points=* ]]; then
-		access_points="${arg#*=}"
-	fi
-	if [[ $arg == --local_mount=* ]]; then
-		local_mount="${arg#*=}"
-	fi
-	if [[ $arg == --mount_options=* ]]; then
-		mount_options="${arg#*=}"
-		mount_options="--${mount_options//,/ --}"
-	fi
-done
-
 # Edit agent config
 daos_config=/etc/daos/daos_agent.yml
-sed -i "s/#.*transport_config/transport_config/g" $daos_config
-sed -i "s/#.*allow_insecure:.*false/  allow_insecure: true/g" $daos_config
-sed -i "s/.*access_points.*/access_points: $access_points/g" $daos_config
+
+# rewrite $daos_config from scratch
+mv $${daos_config} $${daos_config}.orig
+
+exclude_fabric_ifaces=""
+# Get interface names with "s0f0" suffix
+if ifconfig -a | grep 's0f0'; then
+	sof0_interfaces=$(ifconfig -a | grep 's0f0:' | awk '{print $1}' | tr ':' '\n' | grep -v '^$' | awk '!a[$0]++' | sed 's/^/"/g' | sed 's/$/"/g' | paste -sd, -)
+
+	# Append the sof0_interfaces to the existing list
+	exclude_fabric_ifaces="lo,$sof0_interfaces"
+fi
+
+cat > $daos_config <<EOF
+access_points: ${access_points}
+log_file: /var/log/daos_agent/daos_agent.log
+transport_config:
+  allow_insecure: true
+$exclude_fabric_ifaces
+${daos_agent_config}
+
+EOF
 
 # Get names of network interfaces not in first PCI slot
 # The first PCI slot is a standard network adapter while remaining interfaces
 # are typically network cards dedicated to GPU or workload communication
 if [[ "$OS_ID" == "debian" ]]; then
```

Comment:
Will this be needed now ?

Reply:
good catch, this is in the oven for too long :-)

Reply:
Sorry about that. Its mainly from my side.